src/plugins/discover/public/application/view_alert/view_alert_route.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useEffect, useMemo } from 'react';
+import { useHistory, useLocation, useParams } from 'react-router-dom';
+import { sha256 } from 'js-sha256';
+import { i18n } from '@kbn/i18n';
+import { ToastsStart } from 'kibana/public';
+import type { Alert } from '../../../../../../x-pack/plugins/alerting/common';
+import type { AlertTypeParams } from '../../../../../../x-pack/plugins/alerting/common';
+import { getTime, SerializedSearchSourceFields } from '../../../../data/common';
+import type { Filter, TimeRange } from '../../../../data/public';
+import { MarkdownSimple, toMountPoint } from '../../../../kibana_react/public';
+import { DiscoverAppLocatorParams } from '../../locator';
+import { useDiscoverServices } from '../../utils/use_discover_services';
+
+interface SearchThresholdAlertParams extends AlertTypeParams {
+  searchConfiguration: SerializedSearchSourceFields;
+}
+
+interface QueryParams {
+  from: string | null;
+  to: string | null;
+  checksum: string | null;
+}
+
+type NonNullableEntry<T> = { [K in keyof T]: NonNullable<T[keyof T]> };
+
+const LEGACY_BASE_ALERT_API_PATH = '/api/alerts';
+const DISCOVER_MAIN_ROUTE = '/';
+
+const displayError = (title: string, errorMessage: string, toastNotifications: ToastsStart) => {
+  toastNotifications.addDanger({
+    title,
+    text: toMountPoint(<MarkdownSimple>{errorMessage}</MarkdownSimple>),
+  });
+};
+
+const displayRuleChangedWarn = (toastNotifications: ToastsStart) => {
+  const warnTitle = i18n.translate('discover.viewAlert.alertRuleChangedWarnTitle', {
+    defaultMessage: 'Alert rule has changed',
+  });
+  const warnDescription = i18n.translate('discover.viewAlert.alertRuleChangedWarnDescription', {
+    defaultMessage: `Displayed documents might not match the documents triggered notification, 
```

Comment:
I think more precise would be `The displayed documents might not match the documents triggered the alert because the rule configuration changed.`

Reply:
Good suggestion. Be sure to include the word "that" before triggered.

> The displayed documents might not match the documents that triggered the alert because the rule configuration changed.